### PR TITLE
Added travis CI

### PR DESCRIPTION
**Pull request by gluck**
In case you're interested in having mono CI for Cecil, here's the travis conf you can use.
Mono.Cecil.nunit change is compatible with Windows (the old version wasn't working on linux).

Note that there are ~25 UT failures on mono/linux, so enabling travis on Cecil will bring red'ish stickers (rather than green-ish) on PRs.

I also suggest the appveyor config gets commited to the repo in the same way, it's more sane when working with branches and provides a changelog for build configs. You can go to appveyor settings, select "export YAML".

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: csharp
+solution: Mono.Cecil.sln
+install:
+  - nuget restore Mono.Cecil.sln
```
Comment (jbevain) [inline .travis.yml]: Is this needed? We don't reference nugets in the csprojs.

Comment (gluck) [inline .travis.yml]: Then it's not (but it doesn't fail either, and will be required if one day Cecil decides that some NuGet is worth using).
Travis calls it by default in the install step, so once you override the step (to install NUnit runner), you need to call it explicitely.
TL;DR it can be removed

Comment (gluck) [inline .travis.yml]: (more likely than Cecil using a NuGet package would be to grab NUnit as a NuGet package in UTs, it'll be required then as well)

**Comment by jbevain** [comment]
On the subject of tests on Mono, we recently worked on reducing the failures: https://github.com/jbevain/cecil/issues/209

**Comment by gluck** [comment]
Nice, it seems that there are much less that when I tested last time, you can see the build on my travis setup:
https://travis-ci.org/gluck/cecil/builds/64917435

**Comment by jbevain** [comment]
SimpleInterfaces should now be fixed. The last failing test looks indeed bonkers in terms of IL.

**Comment by jbevain** [comment]
And we should be green on Mono now as well.

**Comment by gluck** [comment]
rebased the PR so that Travis tells us the truth:
https://travis-ci.org/gluck/cecil/builds/64933635

**Comment by gluck** [comment]
Still some failures, though they don't seem to always occur.
The [first run](https://travis-ci.org/gluck/cecil/builds/64917435) on this PR didn't have those.

**Comment by jbevain** [comment]
@gluck they're a result of #225, it's fixed in HEAD.

**Comment by gluck** [comment]
Yet another rebase then, we'll see

**Comment by gluck** [comment]
https://travis-ci.org/gluck/cecil/builds/64949521
:tada:
